tests/integration/exec_test.py

```diff
@@ -88,6 +93,66 @@ def test_exec_command_streaming(self):
             res += chunk
         self.assertEqual(res, b'hello\nworld\n')
 
+    def test_exec_start_socket(self):
+        if not api_test.exec_driver_is_native():
+            pytest.skip('Exec driver not native')
+
+        container = self.client.create_container(BUSYBOX, 'cat',
+                                                 detach=True, stdin_open=True)
+        id = container['Id']
+        self.client.start(id)
+        self.tmp_containers.append(id)
```

Comment:
this `self.tmp_containers.append(id)` line should come before `self.client.start(id)` so that the container gets cleaned up if start fails.